src/solve.jl

```diff
@@ -7,7 +7,7 @@ function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase
 end
 
 function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.AbstractDAEProblem},
-                           alg::Union{OrdinaryDiffEqAlgorithm,DAEAlgorithm},
```

Comment:
why the name change?

Reply:
Because we are defining `alg` below in an if statement. The type of `alg` was not inferred properly previously.

Reply:
Maybe one should rather rename `alg` defined below in the function body? I would assume that this requires fewer renamings and leads to a more "natural" function signature?

Reply:
I don't mind either. What we have here actually does less renaming. The function is huge!

Reply:
Renaming `alg` in the `if` block below to `_alg` requires 20 renamings, that's fewer than occurrences of `_alg` in this PR.

Reply:
Ok I recounted, and they are very close but your suggestion has a few less renames. I will change it to have a "natural" function signature.